spec/src/main/asciidoc/core/spi_lite.asciidoc

```diff
@@ -1,12 +1,386 @@
+    ClassConfig addAnnotation(Annotation annotation);
+    ClassConfig removeAnnotation(Predicate<AnnotationInfo> predicate);
+    ClassConfig removeAllAnnotations();
+
+    Collection<MethodConfig> constructors();
+    Collection<MethodConfig> methods();
+    Collection<FieldConfig> fields();
+}
+
+public interface ClassInfo extends DeclarationInfo {
+    String name();
+    String simpleName();
+    PackageInfo packageInfo();
+    List<TypeVariable> typeParameters();
+
+    Type superClass();
+    ClassInfo superClassDeclaration();
+    List<Type> superInterfaces();
+    List<ClassInfo> superInterfacesDeclarations();
+
+    boolean isPlainClass();
+    boolean isInterface();
+    boolean isEnum();
+    boolean isAnnotation();
+    boolean isRecord();
+    boolean isAbstract();
+    boolean isFinal();
+    int modifiers();
+
+    Collection<MethodInfo> constructors();
+    Collection<MethodInfo> methods();
+    Collection<FieldInfo> fields();
+    Collection<RecordComponentInfo> recordComponents();
+}
+----
+
+When the extension method declares a parameter of type `ClassConfig` or `ClassInfo`, it will be called for each discovered class matching the criteria defined on the `@Enhancement` annotation.
+It is possible to navigate to constructors, methods and fields from a `ClassConfig` and configure them.
+
+[source, java]
+----
+public interface MethodConfig extends DeclarationConfig {
+    MethodInfo info();
+
+    MethodConfig addAnnotation(Class<? extends Annotation> annotationType);
+    MethodConfig addAnnotation(AnnotationInfo annotation);
+    MethodConfig addAnnotation(Annotation annotation);
+    MethodConfig removeAnnotation(Predicate<AnnotationInfo> predicate);
+    MethodConfig removeAllAnnotations();
+
+    List<ParameterConfig> parameters();
+}
+
+public interface ParameterConfig extends DeclarationConfig {
+    ParameterInfo info();
+
+    ParameterConfig addAnnotation(Class<? extends Annotation> annotationType);
+    ParameterConfig addAnnotation(AnnotationInfo annotation);
+    ParameterConfig addAnnotation(Annotation annotation);
+    ParameterConfig removeAnnotation(Predicate<AnnotationInfo> predicate);
+    ParameterConfig removeAllAnnotations();
+}
+
+public interface MethodInfo extends DeclarationInfo {
+    String name();
+    List<ParameterInfo> parameters();
+    Type returnType();
+    Type receiverType();
+    List<Type> throwsTypes();
+    List<TypeVariable> typeParameters();
+
+    boolean isConstructor();
+    boolean isStatic();
+    boolean isAbstract();
+    boolean isFinal();
+    int modifiers();
+
+    ClassInfo declaringClass();
+}
+
+public interface ParameterInfo extends DeclarationInfo {
+    String name();
+    Type type();
+
+    MethodInfo declaringMethod();
+}
+----
+
+When the extension method declares a parameter of type `MethodConfig` or `MethodInfo`, it will be called for each method and constructor of each discovered class matching the criteria defined on the `@Enhancement` annotation.
```

Comment:
`When the extension method` -> `When an extension method`?
Not sure but `the` doesn't sound right since you are not referring to one specific method. If you choose to change it, note that it is repeated several times in this file.